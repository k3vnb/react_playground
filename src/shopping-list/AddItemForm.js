import React from 'react'


class AddItemForm extends React.Component {
    onSubmitForm = e => {
        e.preventDefault()
        this.props.onAddItem(e.target.itemToAdd.value)
    }
    render(){
        return (
            <form action="" onSubmit={this.onSubmitForm}>
                <input 
                    name='itemToAdd'
                    type="text"
                    placeholder="carrots"
                    aria-label="Shopping list item"
                />
                <button type="submit">Add Item</button>
            </form>
        );
    }
}

export default AddItemForm;