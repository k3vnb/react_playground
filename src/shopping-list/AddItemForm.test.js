import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddItemForm from './AddItemForm'

describe('<AddItemForm /> Component', () => {
    it('renders complete form', () => {
        expect(toJson(shallow(<AddItemForm />))).toMatchSnapshot()
    })
})
