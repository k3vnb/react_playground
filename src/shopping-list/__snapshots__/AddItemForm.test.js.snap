// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddItemForm /> Component renders complete form 1`] = `
<form
  action=""
>
  <input
    aria-label="Shopping list item"
    placeholder="carrots"
    type="text"
  />
  <button
    type="submit"
  >
    Add Item
  </button>
</form>
`;
